core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -427,6 +446,15 @@ class BlockManagerMasterEndpoint(
       locations.remove(blockManagerId)
     }
 
+    if (storageLevel.useDisk && externalShuffleServiceEnabled) {
```

Comment:
`useDisk` doesn't necessarily mean the data will be on disk (it can be still in memory if there's room). Would that situation cause any problems?

Reply:
No, as here the storage level reflects the actual state and not the desired state. 

Details:
As `UpdateBlockInfo` message only created by `org.apache.spark.storage.BlockManagerMaster#updateBlockInfo` which only called by `org.apache.spark.storage.BlockManager#tryToReportBlockStatus` which called by:
1) `org.apache.spark.storage.BlockManager#reportAllBlocks`
2) `org.apache.spark.storage.BlockManager#reportBlockStatus`

In 1) you can see within the same method that `BlockStatus` is created by `org.apache.spark.storage.BlockManager#getCurrentBlockStatus` where the `useDisk` property only true if `diskStore` really contains the block:

https://github.com/apache/spark/blob/3e7797aa02a48e653b3c167a62cba6512a1d3bcb/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L693-L697

For 2) there is a comment which says here explicitly the storage level reflects the actual and not the desired state: 
https://github.com/apache/spark/blob/3e7797aa02a48e653b3c167a62cba6512a1d3bcb/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L644-L653

But I checked the calls of `org.apache.spark.storage.BlockManager#reportBlockStatus` too and the `BlockStatus` is either `BlockStatus.empty` or computed by  `org.apache.spark.storage.BlockManager#getCurrentBlockStatus` which is analysed in point 1).